src/clients/errors.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * @import { TxOutputId } from "@helios-lang/ledger"
+ */
+
+/**
+ * @typedef {Object} ExpiryDetails
+ * @property {true} isExpired
+ * @property {true} isPremature
+ */
+
+/**
+ * Detects if the tx is not submittable due to validity interval.
+ * @param {string} message
+ * @remarks
+ * Note that different clients may not be able to distinguish between
+ * expired txs and txs that are not yet valid.
+ *
+ * Client code getting this error with the `isIndeterminate` flag may need to query
+ * through the client for current-slot/time to determine if the tx is expired
+ * or premature.
+ *
+ * The `isDeterminate` flag is set by default; client implementations should use the
+ * `expired()` or `premature()` method to indicate a definite state if available.
+ * @extends {Error}
+ */
+export class SubmissionExpiryError extends Error {
+    kind = "SubmissionExpiryError"
+    /**
+     * If true, the tx is definitely expired.  If not, it may be either expired or premature.
+     * @type {boolean | undefined}
+     */
+    isExpired
+
+    /**
+     * If true, the tx is definitely premature.  If undefined, it may be either expired or premature.
+     * @type {boolean | undefined}
+     */
+    isPremature
+
+    /**
+     * If true, the client does not have enough information to determine if the tx is expired or premature.
+     * @type {boolean}
+     */
+    isIndeterminate
+
+    /**
+     * @argument {string} message
+     * @argument {ExpiryDetails} [options]
```

Comment:
I didn't know `argument` could be used instead of `@param

Reply:
Ok, let's keep it consist, and at some point in the future we can change all `@param` tags to `@arg` 

(the abbreviated `@arg` form is better I think, but it appears almost everywhere)

Reply:
fixed